adijif/clocks/ad9545.py

```diff
@@ -0,0 +1,330 @@
+                            APLL_M = self.model.Var(integer=True, lb=1, ub=255)
+
+                            equations = equations + [self.config[n_name] == DPLL_N * APLL_M]
+
+        elif self.solver == "CPLEX":
+            """ Add divider as variables to the model """
+            for i in range(0,4):
+                self.config["r" + str(i)] = exp.integer_var(domain=(self.R_min, self.R_max))
+                self.config["input_ref_" + str(i)] = exp.CpoValue(int(input_refs[i]), type=ctg.Type_Int)
+
+            """ Add PLL N as variables to the model for each PLL profile """
+            for i in range(0,2):
+                if self.PLL_used[i]:
+                    self.config["PLL" + str(i) + "_rate"] = exp.integer_var(domain=(int(self.PLL_out_min[i]), int(self.PLL_out_max[i])))
+
+                for j in range(0,4):
+                    if input_refs[j] != 0:
+                        n_name = "n" + str(i) + "_profile_" + str(j)
+                        self.config[n_name] = exp.integer_var(domain=(self.N_min, self.N_max))
+
+                        """ Internally the PLL block is composed of a Digital PLL and an Analog PLL
+                            with different constraints on dividers.
+                        """
+                        DPLL_N = exp.integer_var(domain=(1, 300000000))
+                        APLL_M = exp.integer_var(domain=(1, 255))
+
+                        equations = equations + [self.config[n_name] == DPLL_N * APLL_M]
+        else:
+            raise Exception("Unknown solver {}".format(self.solver))
+
+        for i in range(0, 4):
+            if self.input_refs[i] != 0:
+                if self.solver == "gekko":
+                    self.config["PLL_in_rate_" + str(i)] = self.model.Intermediate(self.config["input_ref_" + str(i)] / self.config["r" + str(i)])
+                elif self.solver == "CPLEX":
+                    self.config["PLL_in_rate_" + str(i)] = exp.integer_var(domain=(self.PLL_in_min, self.PLL_in_max))
+                    equations = equations + [self.config["PLL_in_rate_" + str(i)] == self.config["input_ref_" + str(i)] / self.config["r" + str(i)]]
+                else:
+                    raise Exception("Unknown solver {}".format(self.solver))
+
+                """ Need to make sure here the PLLs do not receive more than 200 kHz input """
+                equations = equations + [self.config["PLL_in_rate_" + str(i)] < self.PLL_in_max]
+
+        for i in range(0, 2):
+            for j in range(0, 4):
+                if self.input_refs[j] != 0 and self.PLL_used[i]:
+                    n_name = "n" + str(i) + "_profile_" + str(j)
+                    equations = equations + [self.config[n_name] * self.config["PLL_in_rate_" + str(j)] == self.config["PLL" + str(i) + "_rate"]]
+
+        self._add_equation(equations)
+
+        """ Instruct solver to try to avoid Minimum and Maximum PLL rates """
+        for i in range(0, 2):
```

Comment:
Please add a flag to disable optimizations/cost functions. For larger problems, these need to get converted to lexicographic versions